### PR TITLE
small coherence cleanup

### PR DESCRIPTION
**Pull request by lcnr**
r? @eddyb

### CODE REVIEW

File: src/librustc_trait_selection/traits/coherence.rs [hunk 1]
```diff
@@ -389,7 +389,7 @@ fn orphan_check_trait_ref<'tcx>(
     ) -> Vec<Ty<'tcx>> {
         // FIXME(eddyb) figure out if this is redundant with `ty_is_non_local`,
         // or maybe if this should be calling `ty_is_non_local_constructor`.
```
Comment (lcnr) [inline src/librustc_trait_selection/traits/coherence.rs:391]: This `FIXME` is interesting.

Will have to do something about this before we merge this PR

Comment (lcnr) [reply src/librustc_trait_selection/traits/coherence.rs:391]: Looking into the current impl it seems somewhat overly complicated but can't quite think of a clear solution here.

I do feel like using a `TypeWalker` in `fn orphan_check_trait_ref` might actually be the cleanest approach here, but iirc we wanted to slowly phase out uses of this struct.

What's the current status here?

File: src/librustc_trait_selection/traits/coherence.rs [hunk 2]
```diff
@@ -493,8 +489,7 @@ fn def_id_is_local(def_id: DefId, in_crate: InCrate) -> bool {
     }
 }
 
-// FIXME(eddyb) this can just return `bool` as it always returns `Some(ty)` or `None`.
-fn ty_is_non_local_constructor(ty: Ty<'_>, in_crate: InCrate) -> Option<Ty<'_>> {
+fn ty_is_local_constructor(ty: Ty<'_>, in_crate: InCrate) -> bool {
     debug!("ty_is_non_local_constructor({:?})", ty);
```
Comment (nikomatsakis) [inline src/librustc_trait_selection/traits/coherence.rs:496]: Nit: debug! is out of date.

File: src/librustc_trait_selection/traits/coherence.rs [hunk 3]
```diff
@@ -435,20 +435,16 @@ fn orphan_check_trait_ref<'tcx>(
     Err(OrphanCheckErr::NonLocalInputType(non_local_spans))
 }
 
+// FIXME: Return a `Vec` without `Option` here.
 fn ty_is_non_local(tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, in_crate: InCrate) -> Option<Vec<Ty<'tcx>>> {
```
Comment (nikomatsakis) [inline src/librustc_trait_selection/traits/coherence.rs]: Pre-existing, but can you add a comment explaining what the return value is here?

File: src/librustc_trait_selection/traits/coherence.rs (hunk 1 above)
Comment (lcnr) [reply src/librustc_trait_selection/traits/coherence.rs:391]: I really would like to clean this up before merging this.

If we merge it without fixing this, we should at least update the method names :sweat_smile:

**Comment by lcnr** [comment]
@nikomatsakis this should be ready for merge/rereview. Wasn't able to improve `orphan_check_trait_ref` so I just updated the fixme there

**Pull request merged**